parsons/dbt/dbt.py

```diff
@@ -0,0 +1,268 @@
+                which can be used in your dbt profile
+                Not requried if the `REDSHIFT_USERNAME`
+                environment variable set.
+            password: Optional[str]
+                Populates an environment variable REDSHIFT_PASSWORD
+                which can be used in your dbt profile
+                Not requried if the `REDSHIFT_PASSWORD`
+                environment variable set.
+            host: Optional[str]
+                Populates an environment variable REDSHIFT_HOST
+                which can be used in your dbt profile
+                Not requried if the `REDSHIFT_HOST`
+                environment variable set.
+            port: Optional[str]
+                Populates an environment variable REDSHIFT_PORT
+                which can be used in your dbt profile
+                Not requried if the `REDSHIFT_PORT`
+                environment variable set.
+            db: Optional[str]
+                Populates an environment variable REDSHIFT_DB
+                which can be used in your dbt profile
+                Not requried if the `REDSHIFT_DB`
+                environment variable set.
+            raise_errors: bool
+                Default value: False
+                A flag indicating whether errors encountered by
+                the dbt command should be raised as exceptions.
+            slack_channel: Optional[str]
+                If set, will be used to send log results. Can be set
+                with environment variable `SLACK_CHANNEL`
+            slack_webhook: Optional[str]
+                If set, will be used to send log results. Only one
+                of slack_webhook or slack_api_key is necessary.
+                Can be set with environment variable `SLACK_WEBHOOK`
+            slack_api_key: Optional[str]
+                If set, will be used to send log results. Only one
+                of slack_webhook or slack_api_key is necessary.
+                Can be set with environment variable `SLACK_API_KEY`
+        """
+        self.commands = commands
+        self.dbt_schema = check_env.check("DBT_SCHEMA", dbt_schema)
+        self.username = check_env.check(username, "REDSHIFT_USERNAME")
+        self.password = check_env.check(password, "REDSHIFT_PASSWORD")
+        self.host = check_env.check(host, "REDSHIFT_HOST")
+        self.port = check_env.check(port, "REDSHIFT_PORT")
+        self.db = check_env.check(db, "REDSHIFT_DB")
+        self.dbt_project_directory = dbt_project_directory
+        self.raise_errors = raise_errors
+        self.dbt_logger = dbtLogger(
+            slack_channel=slack_channel or os.environ.get("SLACK_CHANNEL"),
+            slack_webhook=slack_webhook or os.environ.get("SLACK_WEBHOOK"),
+            slack_api_key=slack_api_key or os.environ.get("SLACK_API_KEY"),
+        )
+
+    def run(self) -> None:
+        for command in self.commands:
+            self.dbt_command(command)
+        self.dbt_logger.send_to_slack()
```

Comment:
We'd also want this line to be generic, like `send_notification` instead of Slack specifically.